realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1045,23 +1045,33 @@ private Scanner getFullStringScanner(InputStream in) {
         if (!RealmObject.isManaged(parentObject) || !RealmObject.isValid(parentObject)) {
             throw new IllegalArgumentException("Only valid, managed objects can be a parent to an embedded object.");
         }
+
         RealmObjectProxy proxy = (RealmObjectProxy) parentObject;
         long parentPropertyColKey = schema.getSchemaForClass(parentObject.getClass()).getColumnKey(parentProperty);
         RealmFieldType parentPropertyType = schema.getSchemaForClass(parentObject.getClass()).getFieldType(parentProperty);
+        RealmSchema schema = getSchema();
+        RealmObjectSchema parentObjectSchema = schema.getSchemaForClass(parentObject.getClass());
+        String className = schema.getSchemaForClass(clazz).getClassName();
+        String linkedType = parentObjectSchema.getPropertyType(parentProperty);
```

Comment:
What happens if the parentPropotype is not a link? There seems to be a missing test for it?